app/models/feature.rb

```diff
@@ -1,170 +1,54 @@
-class Feature < ActiveRecord::Base
-  belongs_to :spatial_model, :polymorphic => :true, :autosave => false
+class Feature < AbstractFeature
+  class_attribute :automatically_refresh_aggregate
+  self.automatically_refresh_aggregate = true
 
-  attr_writer :make_valid
+  class_attribute :lowres_precision
+  self.lowres_precision = 5
 
-  FEATURE_TYPES = %w(polygon point line)
+  has_one :aggregate_feature, lambda { |feature| aggregated.where(:spatial_model_type => feature.spatial_model_type) }, :foreign_key => :spatial_model_id, :primary_key => :spatial_model_id
 
-  before_validation :sanitize_feature_type
-  validates_presence_of :geog
-  validate :validate_geometry
   validates_inclusion_of :feature_type, :in => FEATURE_TYPES
-  before_save :sanitize
-  after_save :cache_derivatives
 
-  def self.cache_key
-    "#{maximum(:id)}-#{count}"
-  end
+  after_save :refresh_aggregate, if: :automatically_refresh_aggregate?
```

Comment:
```suggestion
  after_create :refresh_aggregate, if: :automatically_refresh_aggregate?
```

Reply:
So after thinking about this a bit I realized that it is important to trigger `after_save`, not `after_create` for the same reason I have the `spatial_model_id?` check in the refresh method. If we've saved a `Feature` and then assigned it to a record we will trigger an update where we do need to refresh the aggregate feature.